TMGToolbox/src/assignment/preprocessing/remove_redundant_functions.py

```diff
@@ -0,0 +1,296 @@
+    
+    version = '0.0.1'
+    tool_run_msg = ""
+    number_of_tasks = 1 # For progress reporting, enter the integer number of tasks here
+    
+    # Tool Input Parameters
+    #    Only those parameters neccessary for Modeller and/or XTMF to dock with
+    #    need to be placed here. Internal parameters (such as lists and dicts)
+    #    get intitialized during construction (__init__)
+    
+    xtmf_ScenarioNumber = _m.Attribute(int) # parameter used by XTMF only
+    Scenario = _m.Attribute(_m.InstanceType) # common variable or parameter
+    ExportFile = _m.Attribute(str)
+
+
+    def __init__(self):
+        #---Init internal variables
+        self.TRACKER = _util.ProgressTracker(self.number_of_tasks) #init the ProgressTracker
+        
+        #---Set the defaults of parameters used by Modeller
+        self.Scenario = _M.scenario #Default is primary scenario
+    
+    def page(self):
+        pb = _tmgTPB.TmgToolPageBuilder(self, title="Remove Redundant Functions from Database v%s" %self.version,
+                     description="Compare the functions used in the networks with those stored in the database. All \
+                     redundant functions (i.e., not used in any scenario) will be exported as an individual file \
+                     and will be removed from the database.",
+                     branding_text="- TMG Toolbox")
+        
+        if self.tool_run_msg != "": # to display messages in the page
+            pb.tool_run_status(self.tool_run_msg_status)
+            
+#        pb.add_select_scenario(tool_attribute_name='Scenario',
+#                               title='Scenario:',
+#                               allow_none=False)
+
+        pb.add_select_file(tool_attribute_name= 'ExportFile',
+                           window_type= 'save_file',
+                           file_filter= "*.txt \n All files (*.*)",
+                           title= "Export Redundant Functions")
+
+        return pb.render()
+    
+    ##########################################################################################################
+        
+    def run(self):
+        self.tool_run_msg = ""
+        self.TRACKER.reset()
+        
+        try:
+            self._Execute()
+        except Exception as e:
+            self.tool_run_msg = _m.PageBuilder.format_exception(
+                e, _traceback.format_exc(e))
+            raise
+        
+        self.tool_run_msg = _m.PageBuilder.format_info("Run complete.")
+    
+    def __call__(self, xtmf_ScenarioNumber, ExportFile):
+        
+        self.Scenario = _m.Modeller().emmebank.scenario(xtmf_ScenarioNumber)
+        self.ExportFile = ExportFile
+
+        if (self.Scenario is None):
+            raise Exception("Scenario %s was not found!" %xtmf_ScenarioNumber)
+        
+        try:
+            self._Execute()
+        except Exception as e:
+            msg = str(e) + "\n" + _traceback.format_exc(e)
+            raise Exception(msg)
```

Comment:
It looks like this code does not belong in the tool.  Why is it taking in a parameters for a scenario number and an export file?

Reply:
Sorry I forgot to change the xtmf part, I was building it from another tool script.
Will change it.